src/mito2/src/read/projection.rs

```diff
@@ -203,21 +230,161 @@ enum BatchIndex {
     Field(usize),
 }
 
+/// Gets a vector with repeated values from specific cache or creates a new one.
+fn repeated_vector_with_cache(
+    data_type: &ConcreteDataType,
+    value: &Value,
+    num_rows: usize,
+    cache_manager: &CacheManager,
+) -> common_recordbatch::error::Result<VectorRef> {
+    if let Some(vector) = cache_manager.get_repeated_vector(value) {
+        // Tries to get the vector from cache manager. If the vector doesn't
+        // have enough length, creates a new one.
+        match vector.len().cmp(&num_rows) {
+            Ordering::Less => (),
+            Ordering::Equal => return Ok(vector),
+            Ordering::Greater => return Ok(vector.slice(0, num_rows)),
+        }
+    }
+
+    // Creates a new one.
+    let vector = new_repeated_vector(data_type, value, num_rows)?;
+    // Updates cache.
+    if vector.len() <= MAX_VECTOR_LENGTH_TO_CACHE {
+        cache_manager.put_repeated_vector(value.clone(), vector.clone());
+    }
+
+    Ok(vector)
+}
+
 /// Returns a vector with repeated values.
 fn new_repeated_vector(
     data_type: &ConcreteDataType,
-    value: ValueRef,
+    value: &Value,
     num_rows: usize,
 ) -> common_recordbatch::error::Result<VectorRef> {
     let mut mutable_vector = data_type.create_mutable_vector(1);
     mutable_vector
-        .try_push_value_ref(value)
+        .try_push_value_ref(value.as_value_ref())
         .map_err(BoxedError::new)
         .context(ExternalSnafu)?;
     // This requires an additional allocation.
-    // TODO(yingwen): Add a way to create repeated vector to data type.
     let base_vector = mutable_vector.to_vector();
     Ok(base_vector.replicate(&[num_rows]))
```

Comment:
```suggestion
    let base_vector = mutable_vector.to_vector();
    Ok(ConstantVector::new(base_vector, num_rows))
```

Reply:
We can't use constant vectors here. e.g. if the data type of the column is string, then we need to return `StringArray` while selecting the column.